src/NHibernate/Criterion/LikeExpression.cs

```diff
@@ -78,17 +83,28 @@ public override SqlString ToSqlString(ICriteria criteria, ICriteriaQuery criteri
 			else
 				lhs.Add(columns[0]);
 
+			IType propertyType = projection.GetTypes(criteria,criteriaQuery).Single();
+			if (value != null && !(value is System.Type) && !propertyType.ReturnedClass.IsInstanceOfType(value))
+			{
+				throw new QueryException(string.Format(
+											"Type mismatch in {0}: expected type {1}, actual type {2}",
+											criteria.GetType(), propertyType.ReturnedClass, value.GetType()));
```

Comment:
> criteria.GetType(): If I read that correctly, it will always say "Type mismatch in NHibernate.Impl.CriteriaImpl:..." (or "NHibernate.Impl.CriteriaImpl.SubCriteria"), since those are the only two implementations of ICriteria? If so it seems like a non-helpful implementation detail and might just say "... in criteria:..." instead.